src/collector/top_score_collector.rs

```diff
@@ -806,14 +816,14 @@ where
     /// If the document is below the current threshold, it will be ignored.
     #[inline]
     pub fn push(&mut self, feature: Score, doc: D) {
-        if let Some(last_median) = self.threshold.clone() {
-            if feature < last_median {
+        let comparable_doc = ComparableDoc { doc, feature };
+        if let Some(last_median) = self.threshold.as_ref() {
+            if &comparable_doc > last_median {
```

Comment:
Calling this out in particular, as we discussed it downstream:

It is, strangely enough.

The reason this swapped is that `TopNComputer` used to be hardcoded for descending order: when support was added for configuring the order via `ComparableDoc`, this codepath wasn't updated to use that wrapper's `Ord` implementation, and so it continued to only support descending order on the feature.